### ISSUE TITLE
RPMs fail to build due to Python syntax error (RHEL6)

### ISSUE THREAD

**Issue by chrissamuel** [NERSC/shifter#32]
Currently trying to build RPMs from the git repo on our RHEL6 boxes fails with:

```
Byte-compiling python modules...
auth.pyconverters.pydockerhub.pydockerv2.pyimagemngr.pySyntaxError: ('invalid syntax', ('/usr/lib64/python2.6/site-packages/shifter_imagegw/imagemngr.py', 266, 77, "          self.logger.warn('Invalid session on system %s'%{request['system']})\n"))

```
which results in rpmbuild failing with:

```
Processing files: shifter-imagegw-16.04.0pre1-1.x86_64
error: File not found: /tmp/Hc_fdoYaUs/BUILDROOT/shifter-16.04.0pre1-1.x86_64/usr/lib64/python2.7/site-packages/shifter_imagegw
```

The code in question is here:

```
$ cat -n ./imagegw/shifter_imagegw/imagemngr.py | fgrep -B1 -A1 -w 266
   265        if not self.check_session(session,request['system']):
   266            self.logger.warn('Invalid session on system %s'%{request['system']})
   267            raise OSError("Invalid Session")
```

I don't speak Python so I can't comment on what's going on here I'm afraid.

Running the Python interpreter on the file in question reveals a cursor pointing at the final closing curly brace.

```
$ python ./imagegw/shifter_imagegw/imagemngr.py
  File "./imagegw/shifter_imagegw/imagemngr.py", line 266
    self.logger.warn('Invalid session on system %s'%{request['system']})
                                                                      ^
SyntaxError: invalid syntax
```

**Comment by chrissamuel** [comment]
Dropping the curly braces around `request['system']` seems to fix it.

```
-          self.logger.warn('Invalid session on system %s'%{request['system']})
+          self.logger.warn('Invalid session on system %s'%request['system'])

```

**Comment by dmjacobsen** [comment]
I'll try the rpmbuild on a RHEL6 image today, thanks for the repot.  Definitely our python code is almost exclusively tested against python2.7, would be good to make sure the image manager can work in RHEL 6 as well.

**Comment by dmjacobsen** [comment]
Hi Chris,

I've fixed up a number of issues related to centos6 and python2.6 support today.  At least for the basic imagegw and dockerv2 pathway all seems well in centos6 now.

Fixed a couple of syntax-in-python2.6 issues: e7c0ff4978da08e288a39e505f2131de67736271 and 7e29a6af75de579bf8e858a1ac86511c7a9ad744
Fixed a deploy issue:  b43c9585d113b0a5635cb5bfff194d89cda25803
Added some documentation: https://github.com/NERSC/shifter/blob/master/doc/install/centos6.rst

Also, note that in the latest HEAD you'll need to explicitly enable slurm support in the rpmbuild.  See the document linked above.

-Doug
